sherpa/instrument.py

```diff
@@ -296,14 +297,20 @@ def init_kernel(self, kernel):
         # be centered on brightest pixel.
         brightPixel = list(numpy.where(kernel == kernel.max())).pop()
 
-        origin = None
         # if more than one pixel qualifies as brightest, such as const2D
         # use the middle of subkernel -- assumes the user provided center at
         # time of kernel extraction, so that should be middle of subkernel.
+        origin = None
```

Comment:
Not necessary for this PR, but a general question:  is there any impact in running set_origin?  Are we doing several instructions that are only applicable if self.origin is set to None? Would it be more efficient for instance to do something like:

    if self.origin is None:
        # if more than one pixel qualifies as brightest, such as const2D
        # use the middle of subkernel -- assumes the user provided center at
        # time of kernel extraction, so that should be middle of subkernel.
        if (not numpy.isscalar(brightPixel)) and len(brightPixel) != 1:
            self.origin = set_origin(kshape)
        else:
            self.origin = set_origin(kshape, brightPixel)

Reply:
I don't have a good understanding of what's going on with the convolution support, so I don't know the answer to this.

Reply:
understanding what 'center' and 'origin' do is on the first item of my bucket list